### ISSUE TITLE
Management console not working while using kogito-data-index-postgresql

### ISSUE THREAD

**Issue by diogomvasconcelos** [apache/incubator-kie-kogito-apps#1026]
Hi, 

I have a project working with data-index using postgresql but I'm having trouble getting the management console running, every time I try to see which processes are running, in the management console, I get this errror:
![Screenshot_1](https://user-images.githubusercontent.com/67365978/133980316-051a8e04-dad3-4de4-8316-bdc0e9c9d09d.jpg)
In data-index it returns something when I try to do a query, I already initialized the tables present here: https://github.com/kiegroup/kogito-apps/blob/main/data-index/data-index-storage/data-index-storage-postgresql/src/main/resources/data_index_create.sql in the postgres database running on docker
![Screenshot_2](https://user-images.githubusercontent.com/67365978/133980393-3a0d2229-10d9-4ea0-b471-2521047fda7b.jpg)
Here is a copy of my docker-compose:

version: '2.1'

services:
  
  postgres:
    image: "postgres"
    ports:
      - "5432:5432"
    volumes:
      - ./postgres/initialize-databases.sql:/docker-entrypoint-initdb.d/initialize-databases.sql
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  zookeeper:
    container_name: zookeeper
    image: strimzi/kafka:0.20.1-kafka-2.6.0
    command: [
      "sh", "-c",
      "bin/zookeeper-server-start.sh config/zookeeper.properties"
    ]
    ports:
      - "2181:2181"
    environment:
      LOG_DIR: "/tmp/logs"

  kafka:
    image: strimzi/kafka:0.20.1-kafka-2.6.0
    container_name: kafka
    command: [
      "sh", "-c",
      "bin/kafka-server-start.sh config/server.properties --override inter.broker.listener.name=$${KAFKA_INTER_BROKER_LISTENER_NAME} --override listener.security.protocol.map=$${KAFKA_LISTENER_SECURITY_PROTOCOL_MAP} --override listeners=$${KAFKA_LISTENERS} --override advertised.listeners=$${KAFKA_ADVERTISED_LISTENERS} --override zookeeper.connect=$${KAFKA_ZOOKEEPER_CONNECT}"
    ]
    depends_on:
      - zookeeper
    ports:
      - "9092:9092"
    environment:
      KAFKA_BROKER_ID: 0
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_LISTENERS: INTERNAL://kafka:29092,EXTERNAL://kafka:9092
      KAFKA_ADVERTISED_LISTENERS: INTERNAL://kafka:29092,EXTERNAL://localhost:9092
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INTERNAL:PLAINTEXT,EXTERNAL:PLAINTEXT
      KAFKA_INTER_BROKER_LISTENER_NAME: INTERNAL
      KAFKA_AUTO_CREATE_TOPICS_ENABLE: "true"
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      LOG_DIR: "/tmp/logs"

  prometheus:
    image: prom/prometheus:v2.8.0
    container_name: prometheus
    volumes:
      - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml:z
    command:
      - '--config.file=/etc/prometheus/prometheus.yml'
    ports:
      - 9090:9090

  grafana:
    image: grafana/grafana:6.6.1
    container_name: grafana
    depends_on:
      - prometheus
    ports:
      - 3000:3000
    volumes:
      - ./grafana/provisioning/:/etc/grafana/provisioning/:z
    environment:
      - PROMETHEUS_URL=http://prometheus:9090
  
  keycloak:
    container_name: keycloak
    image: quay.io/keycloak/keycloak:15.0.2
    ports:
      - "8480:8080"
    volumes:
      - ./keycloak/kogito-realm.json:/tmp/kogito-realm.json
    environment:
      DB_VENDOR: POSTGRES
      DB_ADDR: postgres
      DB_DATABASE: postgres
      DB_USER: postgres
      DB_SCHEMA: public
      DB_PASSWORD: postgres
      KEYCLOAK_USER: admin
      KEYCLOAK_PASSWORD: admin
      KEYCLOAK_IMPORT: /tmp/kogito-realm.json

  data-index:
    container_name: data-index
    image: quay.io/kiegroup/kogito-data-index-postgresql:${KOGITO_VERSION}
    ports:
      - "8180:8080"
    depends_on:
      kafka:
        condition: service_started
    volumes:
      - ./target/protobuf:/home/kogito/data/protobufs/
    environment:
      QUARKUS_DATASOURCE_JDBC_URL: "jdbc:postgresql://postgres:5432/postgres"
      QUARKUS_DATASOURCE_USERNAME: postgres
      QUARKUS_DATASOURCE_PASSWORD: postgres
      KAFKA_BOOTSTRAP_SERVERS: kafka:29092

  management-console:
    container_name: management-console
    image: quay.io/kiegroup/kogito-management-console:${KOGITO_VERSION}
    ports:
      - 8280:8080
    depends_on:
      - data-index
    volumes:
      - ./svg/:/home/kogito/data/svg/
    environment:
      KOGITO_DATAINDEX_HTTP_URL: http://${DOCKER_GATEWAY_HOST}:8180
      KOGITO_MANAGEMENT_CONSOLE_PROPS: -Dkogito.consoles.keycloak.config.url=http://${DOCKER_GATEWAY_HOST}:8480/auth -Dkogito.consoles.keycloak.config.health-check-url=http://localhost:8480/auth/realms/kogito/.well-known/openid-configuration -Dkogito.svg.folder.path=/home/kogito/data/svg
      SCRIPT_DEBUG: "true" 

Not sure what is leading the management-console to fail or what should I do to fix it. Here is a link to the public repository where I have that docker-compose: 
https://github.com/diogomvasconcelos/management-console-debug

Best Regards 
Diogo

**Comment by radtriste** [comment]
Discussion ongoing on https://groups.google.com/g/kogito-development/c/MT9a-u6vumE/m/WU9hKLLJCwAJ?utm_medium=email&utm_source=footer

**Comment by nmirasch** [comment]
Hi Diogo,

I have been trying your example and I some configuration fixes that  make your example run. Here the changes:

   - DOCKER_GATEWAY_HOST=172.17.0.1 (in case you are using linux) This is needed as space to communication between containers, regarding using localhost could be search inside the container (I changed the .env and added .startServices.sh to set that in other cases)
   - Added the '/graphql' as Pere suggested to let the management console find the dataindex endpoint.
   - Added some dependences to use events
   - Added some properties to application.properties like the 'kogito.persistence.type' and event related configuration
   - And, Finally I've replaced the test.SVG by an export of the process to svg

Here the PR with this changes: https://github.com/diogomvasconcelos/management-console-debug/pull/1

These fixes work for me, I hope this helps! Let me know how  it goes

Neus.
